packages/emitter-framework/src/python/components/python-package.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+  Children,
+  createContext,
+  SourceDirectory,
+  SourceDirectoryContext,
+  useContext,
+} from "@alloy-js/core";
+import { PythonPackageScope } from "../symbols/python-package-scope.js";
+import { InitFile } from "./init-file.jsx";
+import { PythonModule, PythonModuleProps } from "./python-module.jsx";
+
+export interface PythonPackageContext {
+  scope: PythonPackageScope;
+  /** Full package path, e.g src/main/java/me/example/code */
+  path: string;
+  /** Name of package, usually name of this directory */
+  name: string;
+  /** Full qualified name of package, e.g me.example.code */
+  qualifiedName: string;
+}
+
+export const PythonPackageContext = createContext<PythonPackageContext>();
+
+export function usePackage() {
+  return useContext(PythonPackageContext);
+}
+
+/** A Python package is a SourceDirectory with an __init__ file.
+ * Every package will have an __init__.py file.
+ */
+export interface PythonPackageProps {
+  name: string;
+  subpackages?: PythonPackageProps[];
+  modules?: PythonModuleProps[];
+  children?: Children;
+}
+
+export function PythonPackage(props: PythonPackageProps) {
```

Comment:
I may be missing something but, is the idea that subpackages are manually provided? Can this be done automatically?

Reply:
There could probably be a heuristic whereby subnamespaces are treated as subpackages, though I think you'd want a mechanism to override that.